profiling/energy-profiler/daemon.cpp

```diff
@@ -0,0 +1,45 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include "daemon.hpp"
+#include <stdexcept>
+#include <thread>
+
+void Daemon::start() {
+  if (!running_) {
+    running_ = true;
+    thread_  = std::thread(&Daemon::run, this);
+  } else {
+    throw std::runtime_error("Daemon already started");
```

Comment:
We may want to think about how to be consistent in handling errors. In your tool, it appears you're sometimes using exceptions, and sometimes using a boolean to handle errors. I'm not sure if there's a policy at the `kokkos-tools` level to avoid exceptions. Probably best to check e.g. with @JBludau.

Reply:
I would even say we don't want to throw when the daemon is already running. This is not an exception but should just do nothing if the daemon is started.

Reply:
removed the exceptions in https://github.com/kokkos/kokkos-tools/pull/300/commits/45028511a5faeca42e030898aa5975a2dd5a1360

Reply:
Yeah, exceptions aren't really used for Kokkos Tools libraries - and I think not throwing here is the right choice.